cmd/tink-worker/main.go

```diff
@@ -22,35 +23,36 @@ func main() {
 
 	// Here we retry failed connections to tink-server using a randomized interval between attempts.
 	// tink-worker is a daemon process, so we do not exit here.
-	var conn *grpc.ClientConn
+	var creds credentials.TransportCredentials
 	for {
-		creds, err := tink.ObtainServerCreds(flagEnvSettings.TinkServerURL)
+		fmt.Printf("Obtaining tink server creds from %s...\n", flagEnvSettings.TinkServerURL)
+		var err error
+		creds, err = tink.ObtainServerCreds(flagEnvSettings.TinkServerURL)
 		if err == nil {
-			conn, err = tink.EstablishServerConnection(flagEnvSettings.TinkServerGRPCAuthority, creds)
-			if err == nil {
-				break
-			}
-			fmt.Printf("Error establishing gPRC connection to tink-server at %s: %v:", flagEnvSettings.TinkServerGRPCAuthority, err)
-		} else {
-			fmt.Printf("Error obtaining server creds from %s: %v", flagEnvSettings.TinkServerURL, err)
+			break
 		}
+		fmt.Printf("Error obtaining server creds from %s: %v\n", flagEnvSettings.TinkServerURL, err)
 
 		// sleep a randomized amount of time before reconnecting to avoid thundering herds
-		// TODO: we may want to make this configurable via a cmdline option?
 		rand.Seed(time.Now().UnixNano())
 		s := rand.Intn(120) + 1 // 2 minutes (120 seconds), plus one second to avoid a possible zero sleep time
 		fmt.Printf("Sleeping %d seconds before attempting to re-connect...\n", s)
 		time.Sleep(time.Duration(s) * time.Second)
 	}
 
+	conn, err := tink.GetGRPCClientConn(flagEnvSettings.TinkServerGRPCAuthority, creds)
+	if err != nil {
+		fmt.Printf("Error generating gPRC client object: %v\n", err)
```

Comment:
This should be followed by an `os.Exit(1)`.

Reply:
This is now moot since we perform retries in the main control loop.